From 791f75fc8fcfb74c4196b021f79017261b399faf Mon Sep 17 00:00:00 2001
From: zhouliang1 <zhouliang1@eswincomputing.com>
Date: Thu, 15 May 2025 16:29:52 +0800
Subject: [PATCH 532/654] feat:enable ddr ecc for televpu

Changelogs:
1. enable ddr ecc for televpu pcie card and
update the related memory addr

Signed-off-by: zhouliang1 <zhouliang1@eswincomputing.com>
Reviewed-by: Joey Zheng <zhengyu@eswincomputing.com>
Tested-by: Guichao Zhou <zhouguichao@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/vpu7702-pcie.dts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts b/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
index 8f19ec7f1925..3cbfe77eeac1 100644
--- a/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
+++ b/arch/riscv/boot/dts/eswin/vpu7702-pcie.dts
@@ -24,8 +24,8 @@
 #define LSPCLK_FREQ		200000000
 
 /* If wanna enable ECC capability of DDR, should reserve highest zone of 1/8 all space for it */
-#define MEMORY_SIZE_H		0x4
-#define MEMORY_SIZE_L		0x0
+#define MEMORY_SIZE_H		0x3
+#define MEMORY_SIZE_L		0x80000000
 #define CMA_SIZE		0x20000000
 
 #include "eswin-win2030-die0-soc.dtsi"
@@ -110,7 +110,7 @@ g2d_12GB_boundary_reserved_4k {
 
 		mmz_nid_0_part_0 {
 			compatible = "eswin-reserve-memory";
-			reg = <0x3 0x0 0x1 0x80000000>;
+			reg = <0x3 0x0 0x1 0x0>;
 			no-map;
 		};
 
-- 
2.51.0

